File: codewars/outlier/outlier.rb
def FindOutlier arr
  countOdd = 0
  countEven = 0

  arr.each do |num|
    if num.even?
      countEven += 1
    else
      countOdd += 1
    end
  end

  if countEven > countOdd
    arr.each do |num|
      return num if num.odd?
    end
  else
    arr.each do |num|
      return num if num.even?
    end
  end
end

puts FindOutlier([2, 6, 8, -10, 3])

File: codewars/outlier/outlier.js
function FindOutlier(array) {
  let countOdd = 0
  let countEven = 0
  let value = 0

  array.map(num => {
    if(num % 2 == 0) {
      countEven++
    } else {
      countOdd++
    }
  })

  if (countEven > countOdd) {
    array.map((num) => {
      if (num % 2 !== 0) {
        value += num
      }
    })
  } else {
    array.map((num) => {
      if (num % 2 == 0) {
        value += num
      }
    })
  }

  return value
}


console.log(FindOutlier([2, 6, 8, -10, 3]))
